### PR TITLE
Updating vscode.md

### PR DESCRIPTION
**Pull request by srmeier**
Updating `vscode.md` to match what is found in the [mm repository](https://github.com/zeldaret/mm/blob/master/docs/tutorial/vscode.md).

This provides an example cpp configuration JSON which helped my solve an issue I was facing.

### CODE REVIEW

File: docs/tutorial/vscode.md [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+- NumberMonger (convert hex to decimal and vice versa)
+- ~~bracket pair colorizer 2~~ (now obsolete due to VSCode's built-in bracket colouring)
+- Better MIPS Support
 
-You will probably also find it helpful to watch Fig's video, although it was not meant to be an actual introduction. But it does show various useful features of VSCode, quite apart from the decompilation aspect.
 
-Useful stuff to know:
+
+
+## Useful stuff to know:
```
Comment (Dragorn421) [inline docs/tutorial/vscode.md:19]:
```suggestion
## Useful keyboard shortcuts
```

File: docs/tutorial/vscode.md [hunk 2]
```diff
@@ -27,4 +32,53 @@ Useful stuff to know:
...
+
+```jsonc
+{
+    "search.useIgnoreFiles": false,
+    "search.exclude": {
+        "**/.git": true,
+        "baserom/**": true,
+        "build/**": true,
+        "expected/**": true,
+        "nonmatchings/**": true,
```
Comment (Dragorn421) [inline docs/tutorial/vscode.md:81]:
```suggestion
```
may be a mm thing, oot doesn't have this folder and never did afaik

File: docs/tutorial/vscode.md [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
-- better mips support
+## Extensions
 
+There are a number of useful extensions available to make work more efficient:
 
+- C/C++ IntelliSense
+- Clang-Format
+- HexInspector (hover on numbers for float and other info)
+- NumberMonger (convert hex to decimal and vice versa)
+- ~~bracket pair colorizer 2~~ (now obsolete due to VSCode's built-in bracket colouring)
```
Comment (Dragorn421) [inline docs/tutorial/vscode.md:13]:
```suggestion
```
no need to mention it anymore then

**Review by Dragorn421 — CHANGES_REQUESTED** [review]

**Review by Dragorn421 — APPROVED** [review]

**Review by EllipticEllipsis — COMMENTED** [review]
Looks good, but maybe it should move out of the tutorial folder now?

**Comment by srmeier** [comment]
> Looks good, but maybe it should move out of the tutorial folder now?

Where would make more sense?

**Comment by EllipticEllipsis** [comment]
Probably the root of the docs folder for now? We're bound to restructure it soon anyway, if we decide to put actual documentation of the repo itself in there.

**Comment by srmeier** [comment]
Sounds good, moved `vscode.md` to the docs folder.

**Review by EllipticEllipsis — APPROVED** [review]

File: docs/vscode.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            "cppStandard": "${default}" // Only ZAPD uses C++, so doesn't really matter
+        }
+    ],
+    "version": 4
+}
+```
+
+## Settings
+
+Add the following to (or create) the `.vscode/settings.json` file for VSCode to search the gitignored asset and assembly files by default:
```
Comment (EllipticEllipsis) [inline docs/vscode.md:70]: Actually, OoT doesn't do this
```suggestion
Add the following to (or create) the `.vscode/settings.json` file for VSCode to search the gitignored asset files by default:
```

**Comment by srmeier** [comment]
@EllipticEllipsis Updated based on your previous comments.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**